src/pkg/cli/whoami.go

```diff
@@ -18,23 +58,19 @@ func Whoami(ctx context.Context, fabric client.FabricClient, provider client.Pro
 		return "", err
 	}
 
-	accountID := resp.Account
 	if account.AccountID() != "" {
-		accountID = account.AccountID()
+		showData.AccountID = account.AccountID()
 	}
 
-	region := resp.Region
+	showData.Region = resp.Region
 	if account.Region() != "" {
-		region = account.Region()
+		showData.Region = account.Region()
 	}
 
-	if account.Details() != "" {
-		accountID += " (" + account.Details() + ")"
-	}
+	showData.Details = account.Details()
+	showData.Provider = account.Provider().Name()
+	showData.SubscriberTier = pkg.SubscriptionTierToString(resp.Tier)
```

Comment:
Converting types to their respective printable strings feels like it should be the responsibility of the `showAccountInfo` call.